src/skspatial/objects/cylinder.py

```diff
@@ -476,6 +481,80 @@ def to_mesh(self, n_along_axis: int = 100, n_angles: int = 30) -> Tuple[np.ndarr
 
         return X, Y, Z
 
+    @classmethod
+    def best_fit(cls, points: array_like) -> Cylinder:
+        """
+        Return the cylinder of best fit for a set of 3D points.
+
+        The points are assumed to lie close to the cylinder surface.
+
+        The algorithm is not guaranteed to produce a meaningful solution with random points.
+
+        Parameters
+        ----------
+        points : array_like
+             Input 3D points.
+
+        Returns
+        -------
+        Cylinder
+            The cylinder of best fit.
+
+        Raises
+        ------
+        ValueError
+            If the points are not 3D.
+            If there are fewer than six points.
+            If the points are collinear.
+            If the points are coplanar.
+
+        References
+        ----------
+        https://www.geometrictools.com/Documentation/LeastSquaresFitting.pdf
+        https://github.com/xingjiepan/cylinder_fitting
+        https://github.com/CristianoPizzamiglio/py-cylinder-fitting
+
+        Examples
+        --------
+        >>> from skspatial.objects import Cylinder
+
+        >>> points = [[0, 2, 0], [0, -2, 0], [0, 0, 2], [5, 2, 0], [5, -2, 0], [5, 0, 2]]
+        >>> cylinder = Cylinder.best_fit(points)
+
+        >>> cylinder.point.round()
+        Point([0., 0., 0.])
+
+        >>> cylinder.vector.round()
+        Vector([5., 0., 0.])
+
+        >>> cylinder.radius
+        2.0
+
+        """
+        points = Points(points)
+
+        if points.dimension != 3:
+            raise ValueError("The points must be 3D.")
+
+        if points.shape[0] < 6:
+            raise ValueError("There must be at least 6 points.")
+
+        if points.are_collinear():
+            raise ValueError("The points must not be collinear.")
+
+        if points.are_coplanar():
+            raise ValueError("The points must not be coplanar.")
+
+        points_centered, centroid = points.mean_center(return_centroid=True)
+        unit_vector, center, radius, _ = _best_fit(points_centered, centroid)
+        axis = Line(point=center, direction=unit_vector)
+        points_1d = axis.transform_points(points)
+        point_a = axis.project_point(points[np.argmin(points_1d)])
+        length = point_a.distance_point(center) * 2
+        vector_ab = unit_vector * length
```

Comment:
This seems like a complicated procedure to get the length of the cylinder. Is this what the algorithm used?

Reply:
As you can see [here ](https://www.geometrictools.com/Documentation/LeastSquaresFitting.pdf) at page 36, where a summary of the algorithm is shown, the procedure to compute the length of the cylinder is missing.
Using the `transform_points` and `project_point` methods, I have found `point_a`, i.e. the the center of the cylinder base. Then, the length is just given by the `point_a` - `center` distance multiplied by 2, where `center` is the center provided by Eberly's algorithm.